src/Core/Framework/Api/Controller/SalesChannelProxyController.php

```diff
@@ -195,6 +206,51 @@ public function assignCustomer(Request $request, Context $context): Response
         return $response;
     }
 
+    /**
+     * @Since("6.2.0.0")
+     * @Route("/api/_proxy/login-as-customer-token-generate", name="api.proxy.login-as-customer-token-generate", methods={"POST"}, defaults={"_acl"={"api_proxy_switch-customer"}})
+     */
+    public function loginAsCustomerTokenGenerate(Request $request, Context $context): Response
+    {
+        if (!$request->request->has(self::SALES_CHANNEL_ID)) {
+            throw new MissingRequestParameterException(self::SALES_CHANNEL_ID);
+        }
+
+        $salesChannelId = (string) $request->request->get('salesChannelId');
+
+        $salesChannel = $this->fetchSalesChannel($salesChannelId, $context);
+
+        if (!$request->request->has(self::CUSTOMER_ID)) {
+            throw new MissingRequestParameterException(self::CUSTOMER_ID);
+        }
+
+        $customerId = (string) $request->request->get('customerId');
+
+        $this->fetchCustomer($customerId, $context);
+
+        $token = $this->loginAsCustomerTokenGenerator->generate($salesChannelId, $customerId);
+
+        $redirectUrlWithToken = sprintf(
+            '%s/%s',
+            $salesChannel->getDomains()->first()->getUrl(),
```

Comment:
The user should be able to select the URL. It could be also complete different domains 🙈 . Also domains can be empty